app/src/lib/git/for-each-ref.ts

```diff
@@ -41,8 +41,14 @@ export async function getBranches(
   const result = await git(
     ['for-each-ref', `--format=${format}`, ...prefixes],
     repository.path,
-    'getBranches'
+    'getBranches',
+    { successExitCodes: new Set([0, 128]) }
```

Comment:
what if we use the `expectedErrors` option here to indicate more clearly that this is not a success case but an error we want to handle specially?

Reply:
I wanted to use `expectedErrors` here, like this:

```ts
    ...
    'getBranches',
    expectedErrors: new Set([GitError.NotAGitRepository]),
  })
```

But upstream Git fixed this error message in https://github.com/git-for-windows/git/commit/fc045fe7d4fa220f19274715c936636fe0516ea3 which is in `2.17.1`:

```
fatal: Not a git repository (or any of the parent directories): .git
```

to this:

```
fatal: not a git repository (or any of the parent directories): .git
```

So I need to publish a new version of `dugite` to address this - https://github.com/desktop/dugite/pull/199 is the upstream issue.

I'll leave a note here to indicate that this is some future cleanup work, but I'm holding off on upgrading Git to 2.18.0 until I can do some more thorough testing.